internal/encoding/parquet/parquet.go

```diff
@@ -111,13 +112,44 @@ func (i *iterator) Schema() typeof.Schema {
 	for _, c := range schema.Columns() {
 		t := c.Type()
 
-		if t, supported := typeof.FromParquet(t); supported {
-			result[c.Name()] = t
+		if t == nil {
+			p := getSchemaLogicalType(c)
+
+			if t, supported := typeof.FromParquet(&p); supported {
+				result[c.Name()] = t
+			}
+		} else {
+			if t, supported := typeof.FromParquet(t); supported {
+				result[c.Name()] = t
+			}
 		}
 	}
 	return result
 }
 
+func getSchemaLogicalType(c *goparquet.Column) parquet.Type {
```

Comment:
Would rename it to `typeOf` since it's taking a column and returns its type.

Reply:
Added a new method which does what you have recommended

Reply:
This looks unresolved ^

Reply:
@atris please merge the code as I described above :)